### DISCUSSION TITLE
Examples/templates would be super duper nice

### DISCUSSION THREAD

**Discussion by jessekrubin** [egoist/tsup#645] [Ideas]
Imo adding an examples/templates dir for projects of different types would be super nice and helpful. 

I think the following would be good for reference:

- [ ] Node lib
- [ ] Node cli
- [ ] Web lib
- [ ] React lib
- [ ] All of the above + jest/vitest testing (vitest + tsup is (as far as I can tell) a killer combo)

**Comment by jessekrubin** [comment]
PS/FWIW: tsup is alarmingly good. It really does "just work".

**Reply by iongion** [reply]
This, so much this!

I come after a `vite` configuration run of 2 weeks trying to modernize an electron & create-react-app stack.
Which I've failed, due to my own lack of knowledge of course, not due to the tools themselves, they are a huge step forward!

But the thing with `tsup` is that it just fitted like a glove with all my concerns, local private npm packages in a monorepo, shared by both backend (node) and frontend (browser) ... this while still using typescript for both!

I feel I am finally using a compiler toolchain in the traditional way and combined with https://www.npmjs.com/package/@vercel/ncc - bundling has never been better!
